vamp-server/convert.cpp

```diff
@@ -556,8 +558,15 @@ readInput(string format, RequestOrResponse::Direction direction, bool &eof)
     if (format == "json") {
         string err;
         auto result = readInputJson(direction, err, eof);
-        if (err != "") throw runtime_error(err);
-        else return result;
+        const bool isRecognisedError = !result.success && result.errorText != "";
+        
+        // if the RequestOrResponse (result) has been populated with success=false and an error message
+        // then the server returned a well formed error, it is safe to return it for conversion 
+        // -- but if err is populated, something else has gone wrong
+        if (isRecognisedError || err == "") 
```

Comment:
You are correct, and that is exactly what I want to do. 

Perhaps the confusion is that the "but if err is populated..." comment is about falling into the else clause, so shouldn't be above the condition?

By "if err is populated" I mean "not an empty string", which it is on construction - so provided it got through `readInputJson()` without being populated it is safe to assume the input has been read correctly and can be return for further conversion. This is also the case with known errors, because later code is able to output errors, so I don't want to throw at this point. I want anything else to full through to the else clause and throw an exception as before.